main.go

```diff
@@ -212,14 +212,31 @@ func run() int {
 	http.HandleFunc(path.Join(*routePrefix, "/logs"), func(w http.ResponseWriter, r *http.Request) {
 		id, err := strconv.ParseInt(r.URL.Query().Get("id"), 10, 64)
 		if err != nil {
-			http.Error(w, "Invalid probe id", 500)
-			return
+			id = -1
 		}
-		result := rh.Get(id)
-		if result == nil {
-			http.Error(w, "Probe id not found", 404)
+		target := r.URL.Query().Get("target")
+		result := new(prober.Result)
+		if err == nil && target != "" {
+			http.Error(w, "Probe id and target can't be defined at the same time", http.StatusInternalServerError)
+			return
+		} else if id == -1 && target == "" {
```

Comment:
```suggestion
		}
		if id == -1 && target == "" {
```

make this check independent to keep it clean and easy to read.